pkg/operator/ceph/object/controller_test.go

```diff
@@ -754,6 +755,259 @@ func TestCephObjectStoreControllerMultisite(t *testing.T) {
+			opManagerContext:    ctx,
+		}
+
+		_, err := r.context.Clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
+		if !k8serrors.IsAlreadyExists(err) {
+			_, err = r.context.Clientset.CoreV1().Secrets(namespace).Update(ctx, secret, metav1.UpdateOptions{})
+			assert.NoError(t, err)
+		}
+
+		return r
+	}
+
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      store,
+			Namespace: namespace,
+		},
+	}
+
+	currentAndDesiredCephVersion = func(ctx context.Context, rookImage string, namespace string, jobName string, ownerInfo *k8sutil.OwnerInfo, context *clusterd.Context, cephClusterSpec *cephv1.ClusterSpec, clusterInfo *client.ClusterInfo) (*cephver.CephVersion, *cephver.CephVersion, error) {
+		return &cephver.Pacific, &cephver.Pacific, nil
+	}
+
+	t.Run("create an external object store", func(t *testing.T) {
+		objects := []runtime.Object{
+			cephCluster,
+			externalObjectStore,
+			rgwAdminOpsUserSecret,
+		}
+		r := getReconciler(objects)
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.False(t, calledCommitConfigChanges)
+		err = r.client.Get(ctx, req.NamespacedName, externalObjectStore)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete the same external store", func(t *testing.T) {
+		objects := []runtime.Object{
+			cephCluster,
+			externalObjectStore,
+			rgwAdminOpsUserSecret,
+		}
+		r := getReconciler(objects)
+
+		calledCommitConfigChanges = false
+
+		// no dependents
+		dependentsChecked := false
+		cephObjectStoreDependentsOrig := cephObjectStoreDependents
+		defer func() { cephObjectStoreDependents = cephObjectStoreDependentsOrig }()
+		cephObjectStoreDependents = func(clusterdCtx *clusterd.Context, clusterInfo *client.ClusterInfo, store *cephv1.CephObjectStore, objCtx *Context, opsCtx *AdminOpsContext) (*dependents.DependentList, error) {
+			dependentsChecked = true
+			return &dependents.DependentList{}, nil
+		}
+
+		err := r.client.Get(ctx, req.NamespacedName, externalObjectStore)
+		assert.NoError(t, err)
+		externalObjectStore.DeletionTimestamp = &metav1.Time{
+			Time: time.Now(),
+		}
+		err = r.client.Update(ctx, externalObjectStore)
+		assert.NoError(t, err)
+
+		// have to also track the same objects in the rook clientset
+		r.context.RookClientset = rookfake.NewSimpleClientset(externalObjectStore)
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, dependentsChecked)
+		assert.False(t, calledCommitConfigChanges)
+	})
+
+	t.Run("create an external object store with missing secret", func(t *testing.T) {
+		objects := []runtime.Object{
+			cephCluster,
+			externalObjectStore,
+		}
+		r := getReconciler(objects)
+		res, err := r.Reconcile(ctx, req)
+		assert.Error(t, err)
+		assert.False(t, res.Requeue)
+		assert.False(t, calledCommitConfigChanges)
+		err = r.client.Get(ctx, req.NamespacedName, externalObjectStore)
```

Comment:
Sorry, I am a bit confused here, do `get` need to return the error here? Since  `externalObjectStore` reconcile failed. Same for 1003.

Reply:
I think `get` returns `nil` as the error value here since `externalObjectStore` is still present in the runtime environment?

https://github.com/rook/rook/pull/9710/files#diff-10bcb1ddae3b1fddd221dda0765d8dc3987af28dae3a1944dbeaadc31da531fdR979

Reply:
okay

Reply:
What do we need this `get` for?

Reply:
Redundant, removed.